src/cclib/parser/qchemparser.py

```diff
@@ -94,6 +94,17 @@ def before_parsing(self):
             'ALPHA MOLECULAR ORBITAL COEFFICIENTS'
         )
 
+        self.gradient_headers = (
+            'Full Analytical Gradient',
+            'Gradient of SCF Energy',
+            'Gradient of MP2 Energy',
+        )
+
+        self.hessian_headers = (
+            'Hessian of the SCF Energy',
+            'Final Hessian.',
```

Comment:
Yes, I will add some small regression tests; I'd hold off on unit tests simply because we don't have gradients and Hessians implemented for most parsers right now.